### DISCUSSION TITLE
Why does tab complete not work unless you have Essentails permission for the same command?

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#7124] [Q&A]
For example. /warp does not show as a tab complete or an actual command unlesss you have the `essentials.warp` permission, by default essentials says that all other plugins should take over the command. players can use /warp via cmi, it shows the GUI etc, its only the atuo tab just does not work.
This is not just for warp, but ALL commands that conflict with essentials, tpa, mail, all the common ones that people need to use day to day.
Im not sure if this is a bug or if additional setup is required to resolve tab-complete to favor CMI when that is your permission you have.

**Comment by mrfloris** [comment]
CMI does 100% not require essentials.* permissions to work with tab complete, that's probably another plugin controlling these commands, like a command blocker, or permission manager that is setup incorrectly, or multiple plugins using the same permission nodes or even commands.

**Reply by TomLewis** [reply]
No, I know that it does not require essentials permissions, that's not the ticket at all.

 What I'm saying is with both plugins CMI is not taking possession of the tab complete for that command but somehow essentials is.

Let me make this as simple as possible.

/Warp is a command in both CMI and essentials.

Players have /warp permission from CMI. Not essentials.

There is no tab complete for /warp, but they can use the command.

Unless you are op or have the essentials permission, then the tab complete works.

I think you have misread my ticket for poor setup, this is a bug.

**Comment by TomLewis** [comment]
I found a really rubbish fix, in Essentails you have to add each command to the
```
disabled-commands:
 - playtime
 - ping
 - compass
```

Does this mean that the command priority is higher in Essentails than CMI?
